src/backend/api/meals.js

```diff
@@ -1,15 +1,179 @@
 const express = require("express");
+const Knex = require("knex");
+const { groupBy } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
+  const { maxPrice, availableReservations, title, createdAfter, limit } = unpackParams(request); // This step will throw if any params are invalid
+  const reservation = await knex('reservation')
+
+  let meals = knex('meal')
+
+  if (maxPrice != Number.MAX_VALUE) {
+    meals = meals.where('price', '<', maxPrice)
+  }
+  if(title.length!==0){
+    meals=meals.where('title','LIKE',`%${title}%`)
+  }
+  if(createdAfter) {
+    meals=meals.where('created_date','>',createdAfter)
+  }
+if(limit!==Number.MAX_VALUE) {
+  meals=meals.limit(limit)
+}
+
+  
+
+  if (availableReservations==true) {
```

Comment:
No reason to check if it is true unless it has to be exactly `true`
```suggestion
  if (availableReservations) {
```

Reply:
Thanks